lib/validators/post_validator.rb

```diff
@@ -141,6 +142,26 @@ def unique_post_validator(post)
     end
   end
 
+  def force_edit_last_validator(post)
+    return if SiteSetting.max_consecutive_replies == 0 || post.id || post.acting_user.try(:staff?) || private_message?(post)
+
+    topic = post.topic
+
+    last_posts_count = DB.query_single(<<~SQL, topic_id: topic.id, max_replies: SiteSetting.max_consecutive_replies, user_id: post.user_id).first
+      WITH topic_posts AS (
+        SELECT * FROM posts
```

Comment:
```suggestion
        SELECT user_id FROM posts
```

Since that's all you care about ;)

Reply:
I also like this style, but that's up to you

```ruby
DB.query_single(<<~SQL, topic_id: post.topic_id, user_id: post.acting_user.id, max_replies: SiteSetting.max_consecutive_replies).first
  SELECT COUNT(*)
  FROM (
    SELECT user_id
      FROM posts
     WHERE deleted_at IS NULL
       AND NOT hidden
       AND topic_id = :topic_id
     ORDER BY post_number DESC
     LIMIT :max_replies
  ) c
  WHERE c.user_id = :user_id
SQL
```